test/nlp/quasi-newton.jl

```diff
@@ -115,69 +126,46 @@
       Hop = hess_op!(nlp, x, Hv)
       @test Hop * v ≈ H(x) * v
 
-      reset_data!(nlp)
-      Hop = hess_op!(nlp, x, Hv)
-      @test Hop * v == v
+      if QNO ∈ [LSR1Operator, LBFGSOperator]
+        # I am not sure reset! makes sense for the other QN operator since we have to give a value to initialize it
+        # we can add some default values in LinearOperators later if needed, but it is easier to simply update d / σ
```

Comment:
If it is not necessary, couldn't you add a `reset!` which does nothing?
Or implement `NLPModels.reset_data!(nlp::QuasiNewtonModel)` for the two new operators.